### PR TITLE
Extract common code from "read" , "write" and "open"

### PR DESCRIPTION
**Pull request by daynin**

### CODE REVIEW

File: js/files.ts [hunk 1]
```diff
@@ -89,18 +89,41 @@ export async function open(
   return new File(rid);
 }
 
+async function getBaseRes(
+  startOperation: typeof msg.Read.startRead | typeof msg.Write.startWrite,
+  endOperation: typeof msg.Read.endRead | typeof msg.Write.endWrite,
+  addRid: typeof msg.Read.addRid | typeof msg.Write.addRid,
+  innterType: msg.Any.Read | msg.Any.Write,
```
Comment (hayd) [inline js/files.ts]: typo.

**Comment by daynin** [comment]
@hayd I'll try to extract common code for "open"

**Comment by daynin** [comment]
> Can this generalize to all the other ops functions?

@hayd done

**Comment by ry** [comment]
This is more lines of code and more abstractions... Not sure it's a good trade off

**Comment by daynin** [comment]
Yes, but it removes copy-pasted code, and if we'll change something, for example, in getCommonBaseRes it'll applies to all functions

**Comment by daynin** [comment]
It seems to me, it's easier to support this code. @ry what do you think?

**Comment by ry** [comment]
@daynin If you can get it to somehow be a net loss of code - I would be for it. Otherwise I think it adds complexity unnecessarily.

**Pull request closed without merging**